api/views/views.py

```diff
@@ -586,3 +612,69 @@ def post(self, request, format=None):
         except BaseException as e:
             logger.error(str(e))
             return HttpResponse(status=404)
+
+
+class ZipListPhotosView_V2(APIView):
+    def post(self, request):
+        import shutil
+
+        free_storage = shutil.disk_usage("/").free
+        data = dict(request.data)
+        if "image_hashes" not in data:
+            return
+        photo_query = Photo.objects.filter(owner=self.request.user)
+        photos = photo_query.in_bulk(data["image_hashes"])
+        if len(photos) == 0:
+            return
+        total_file_size = photo_query.aggregate(Sum("size"))["size__sum"] or None
+        if free_storage < total_file_size:
+            return Response(data={"status": "Insufficient Storage"}, status=507)
+        file_uuid = uuid.uuid4()
+        filename = str(str(file_uuid) + str(self.request.user.id) + ".zip")
+        user = self.request.user
+        print(user.id)
+        job_id = create_download_job(
+            LongRunningJob.JOB_DOWNLOAD_PHOTOS,
+            user=user,
+            photos=photos,
+            filename=filename,
+        )
+        response = {"job_id": job_id, "url": file_uuid}
+
+        return Response(data=response, status=200)
+
+    def get(self, request):
+        try:
+            job = LongRunningJob.objects.get(job_id=request.data["job_id"])
+            if job.finished:
+                return Response(data={"status": "SUCCESS"}, status=200)
+            elif job.failed:
+                return Response(
+                    data={"status": "FAILURE", "result": job.result}, status=500
+                )
+            else:
+                return Response(
+                    data={"status": "PENDING", "progress": job.result}, status=202
+                )
+        except BaseException as e:
+            logger.error(str(e))
+            return Response(status=404)
+
+
+class DeleteZipView(APIView):
```

Comment:
This can be part of ZipListPhotosView_V2, I think :)

Reply:
oh okay cool then will implement the delete inside the zip view only

Reply:
hey i can do this but then i need the file path so sending it through body wont be good for a  delete request moreover would have to edit the urls.py for make call to a function based view but then the pattern of all class based views would not continue so i was thinking of making it as a separate view class